packages/backend-storage/src/factory.test.ts

```diff
@@ -108,12 +108,4 @@ void describe('AmplifyStorageFactory', () => {
       )
     );
   });
-
-  void it('throws on invalid name', () => {
-    const storageFactory = defineStorage({ name: '!$87++|' });
```

Comment:
Assuming this string makes it's way into logical ids and / or physical ids, there are some limitations on valid values. We either need to relax this validation, but still have validation or we need to sanitize the input value before passing it to underlying components

Reply:
@josefaidt Can you chime in on what validations here?